### PR TITLE
Feature/169 metadata support

### PR DESCRIPTION
**Pull request by vavrajosef**
Closes issue #169

### CODE REVIEW

File: pkg/builder/statefulset_builder.go [hunk 1]
```diff
@@ -156,15 +156,15 @@ func buildStsVolumeClaimTemplates(mariadb *mariadbv1alpha1.MariaDB) []corev1.Per
...
+			Spec: mariadb.Spec.VolumeClaimTemplate.Spec,
 		},
 	}
 	if mariadb.Galera().Enabled {
 		pvcs = append(pvcs, corev1.PersistentVolumeClaim{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: galeraresources.GaleraConfigVolume,
 			},
-			Spec: *mariadb.Galera().VolumeClaimTemplate,
+			Spec: mariadb.Galera().VolumeClaimTemplate.Spec,
```
Comment (mmontes11) [inline pkg/builder/statefulset_builder.go:167]:
```suggestion
func buildStsVolumeClaimTemplates(mariadb *mariadbv1alpha1.MariaDB) []corev1.PersistentVolumeClaim {
	vctpl := mariadb.Spec.VolumeClaimTemplate
	vctpl.ObjectMeta.Name = StorageVolume
	pvcs := []corev1.PersistentVolumeClaim{
		vctpl,
	}
	if mariadb.Galera().Enabled {
		galeraVctpl := *mariadb.Galera().VolumeClaimTemplate
		galeraVctpl.ObjectMeta.Name = galeraresources.GaleraConfigVolume
		pvcs = append(pvcs, galeraVctpl)
	}
	return pvcs
}
```

**Review by mmontes11 — CHANGES_REQUESTED** [review]
Thanks for your contribution, a couple of things:
- We need to add the `metadata` field to the `volumeClaimTemplates`, which is the main purpose of this feature. I've added a commit suggestion for that
- Run `make gen` to generate code and CRDs.  The CI is failing because of this
- To run test locally:
```bash
make cluster
make install
make net
make test
```

Let me know if you have any question!

File: pkg/builder/statefulset_builder.go (hunk 1 above)
Comment (mmontes11) [inline pkg/builder/statefulset_builder.go:167]: can we do something like this?:

```golang
func buildStsVolumeClaimTemplates(mariadb *mariadbv1alpha1.MariaDB) []corev1.PersistentVolumeClaim {
	vctpl := mariadb.Spec.VolumeClaimTemplate
	vctpl.ObjectMeta.Name = StorageVolume
	pvcs := []corev1.PersistentVolumeClaim{
		vctpl,
	}
	if mariadb.Galera().Enabled {
		galeraVctpl := *mariadb.Galera().VolumeClaimTemplate
		galeraVctpl.ObjectMeta.Name = galeraresources.GaleraConfigVolume
		pvcs = append(pvcs, galeraVctpl)
	}
	return pvcs
}
```

**Review by mmontes11 — CHANGES_REQUESTED** [review]

File: pkg/builder/statefulset_builder.go (hunk 1 above)
Comment (vavrajosef) [reply pkg/builder/statefulset_builder.go]: I have applied this suggestion.

**Comment by vavrajosef** [comment]
Thanks for looking into this and thanks for the feedback. I have updated the PR. Please let me know if this is what you had in mind.

**Comment by mmontes11** [comment]
Hey @vavrajosef ! thanks for your changes, they look fine!

However, I've just realized that the [issue](https://github.com/mariadb-operator/mariadb-operator/issues/169) definition implies breaking changes in the CRDs, something we want to avoid whenever possible. What I would suggest is defining our own `VolumeClaimTemplate` type embedding the `corev1.PersistentVolumeClaimSpec ` and adding annotations and labels, which is the main purpose of this issue. It would be something like:

```golang
type VolumeClaimTemplate struct {
    corev1.PersistentVolumeClaimSpec 
    Labels      map[string]string `json:"labels,omitempty"`
    Annotations map[string]string `json:"annotations,omitempty"`
}
``` 
This way, the existing `MariaDB` definitions in our users clusters are still valid and no changes from their side are needed to upgrade.

Sorry for misguiding you! I will be picking your changes to another branch and adding you as a co-author.

**Comment by mmontes11** [comment]
Closed in favor of:

https://github.com/mariadb-operator/mariadb-operator/pull/185

Thanks for your contribution @vavrajosef

**Pull request closed without merging**